core/headerchain.go

```diff
@@ -137,14 +150,22 @@ func (hc *HeaderChain) collectBlockManifest(h *types.Header) (types.BlockManifes
 	if hc.engine.IsDomCoincident(h) {
 		return manifest, nil
 	}
-	// Recursively get the ancestor manifest, until a coincident ancestor is found
-	ancestor := hc.GetHeader(h.ParentHash(), h.NumberU64()-1)
-	if ancestor == nil {
-		return types.BlockManifest{}, errors.New("ancestor not found")
-	}
-	ancManifest, err := hc.collectBlockManifest(ancestor)
-	if err != nil {
-		return nil, errors.New("unable to get manifest for ancestor")
+	var ancManifest types.BlockManifest
+	var err error
+	// First check if the ancestor manifest is cached. Otherwise, collect the ancestor's manifest.
+	if m, ok := hc.manifestCache.Get(h.ParentHash()); ok {
+		ancManifest = m.(types.BlockManifest)
+		ancManifest = append(ancManifest, h.ParentHash())
```

Comment:
Is this redundant with line 170?

Reply:
No, this method is for collecting manifests (lists of block hashes), while the other is for collecting subordinate rollups (lists of ETXs). The structure is similar, though.